src/index/ranker/okapi_bm25.cpp

```diff
@@ -15,24 +16,24 @@ namespace index
 
 const std::string okapi_bm25::id = "bm25";
 
-okapi_bm25::okapi_bm25(double k1, double b, double k3) : k1_{k1}, b_{b}, k3_{k3}
+okapi_bm25::okapi_bm25(float k1, float b, float k3) : k1_{k1}, b_{b}, k3_{k3}
 {
     /* nothing */
 }
 
-double okapi_bm25::score_one(const score_data& sd)
+float okapi_bm25::score_one(const score_data& sd)
 {
-    double doc_len = sd.idx.doc_size(sd.d_id);
+    float doc_len = sd.idx.doc_size(sd.d_id);
 
     // add 1.0 to the IDF to ensure that the result is positive
-    double IDF = std::log(
+    float IDF = fastapprox::fasterlog(
         1.0 + (sd.num_docs - sd.doc_count + 0.5) / (sd.doc_count + 0.5));
```

Comment:
Will this cause an up-conversion to double before down-converting to float for the argument? We may need to say things like `1.0f` and `0.5f` instead.